public/app/features/dashboard/state/DashboardMigrator.ts

```diff
@@ -1480,3 +1484,34 @@ function ensureXAxisVisibility(panel: PanelModel) {
 
   return panel;
 }
+
+function migrateHideFromFunctionality(panel: PanelModel) {
+  // hideFrom.viz implies hideFrom.tooltip
+  if (panel.fieldConfig && panel.fieldConfig.defaults && panel.fieldConfig.defaults.custom) {
+    const hideFrom = panel.fieldConfig.defaults.custom.hideFrom;
+    if (hideFrom) {
+      panel.fieldConfig.defaults.custom.hideFrom = {
+        tooltip: hideFrom.viz === true ? hideFrom.viz : (hideFrom.tooltip ?? false),
+        ...panel.fieldConfig.defaults.custom.hideFrom,
+      };
+    }
+  }
```

Comment:
might not need this? cause this can only be applied as overrides, otherwise you'd be hiding all fields :)